qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -196,60 +238,63 @@ def _build_cr_circuit(
 
         return cr_circuit
 
-    def _build_cr_schedule(
-        self,
-        backend: Backend,
-        flat_top_width: float,
-        sigma: float,
-    ) -> pulse.ScheduleBlock:
+    def _build_cr_schedule(self, flat_top_width: float) -> pulse.ScheduleBlock:
         """GaussianSquared cross resonance pulse.
 
         Args:
-            backend: The target backend.
             flat_top_width: Total length of flat top part of the pulse in units of dt.
-            sigma: Sigma of Gaussian edges in units of dt.
 
         Returns:
             A schedule definition for the cross resonance pulse to measure.
         """
         opt = self.experiment_options
 
         # Compute valid integer duration
-        cr_duration = round_pulse_duration(
-            backend=backend, duration=flat_top_width + 2 * sigma * opt.risefall
-        )
+        duration = flat_top_width + 2 * opt.sigma * opt.risefall
+        valid_duration = int(opt.granularity * np.floor(duration / opt.granularity))
 
-        with pulse.build(backend, default_alignment="left", name="cr") as cross_resonance:
+        with pulse.build(default_alignment="left", name="cr") as cross_resonance:
 
             # add cross resonance tone
             pulse.play(
                 pulse.GaussianSquare(
-                    duration=cr_duration,
+                    duration=valid_duration,
                     amp=opt.amp,
-                    sigma=sigma,
+                    sigma=opt.sigma,
                     width=flat_top_width,
                 ),
-                pulse.control_channels(*self.physical_qubits)[0],
+                pulse.ControlChannel(opt.cr_channel),
             )
             # add cancellation tone
             if not np.isclose(opt.amp_t, 0.0):
                 pulse.play(
                     pulse.GaussianSquare(
-                        duration=cr_duration,
+                        duration=valid_duration,
                         amp=opt.amp_t,
-                        sigma=sigma,
+                        sigma=opt.sigma,
                         width=flat_top_width,
                     ),
-                    pulse.drive_channel(self.physical_qubits[1]),
+                    pulse.DriveChannel(self.physical_qubits[1]),
                 )
             else:
-                pulse.delay(cr_duration, pulse.drive_channel(self.physical_qubits[1]))
+                pulse.delay(valid_duration, pulse.DriveChannel(self.physical_qubits[1]))
 
             # place holder for empty drive channels. this is necessary due to known pulse gate bug.
-            pulse.delay(cr_duration, pulse.drive_channel(self.physical_qubits[0]))
+            pulse.delay(valid_duration, pulse.DriveChannel(self.physical_qubits[0]))
 
         return cross_resonance
 
+    def set_experiment_options(self, **fields):
+        super().set_experiment_options(**fields)
+
+        # Effective length of Gaussian rising falling edges for fit guess (in dt).
```

Comment:
Maybe add to this comment something like: Set analysis option for initial guess that depends on experiment option values.